e2e/test/scenarios/actions/actions-in-object-detail-view.cy.spec.js

```diff
@@ -2,237 +2,191 @@ import moment from "moment-timezone";
+      beforeEach(() => {
+        asAdmin(() => {
+          cy.get("@modelId").then(modelId => {
+            createImplicitActions({ modelId });
+
+            cy.createQuestionAndDashboard({
+              questionDetails: {
+                name: "Score detail",
+                display: "object",
+                database: WRITABLE_DB_ID,
+                query: {
+                  "source-table": `card__${modelId}`,
+                },
+              },
+              dashboardDetails: DASHBOARD,
+            }).then(({ body: { dashboard_id } }) => {
+              cy.wrap(dashboard_id).as("dashboardId");
+            });
           });
         });
+      });
 
+      it("does not show model actions in model visualization on a dashboard", () => {
         asAdmin(() => {
-          cy.log("As admin: verify database actions are enabled");
-          visitModelDetail(modelId);
-          assertActionsTabExists();
-
-          cy.log("As admin: Verify that there are no model actions to run");
-          visitObjectDetail(modelId, FIRST_SCORE_ROW_ID);
-          objectDetailModal().within(() => {
-            assertActionsDropdownNotExists();
+          cy.get("@dashboardId").then(dashboardId => {
+            visitDashboard(dashboardId);
           });
 
-          cy.log("As admin: create basic model actions");
-          createBasicModelActions(modelId);
-
-          cy.log("As admin: verify there are model actions to run");
-          visitObjectDetail(modelId, FIRST_SCORE_ROW_ID);
-          objectDetailModal().within(() => {
-            assertActionsDropdownExists();
+          cy.findByTestId("dashcard").within(() => {
+            assertActionsDropdownNotExists();
           });
         });
+      });
+    });
 
-        asNormalUser(() => {
-          cy.log("As normal user: verify there are model actions to run (1)");
-          visitObjectDetail(modelId, FIRST_SCORE_ROW_ID);
-          objectDetailModal().within(() => {
-            assertActionsDropdownExists();
-          });
-
-          cy.log("As normal user: verify update form gets prefilled");
-          openUpdateObjectModal();
-          actionExecuteModal().within(() => {
-            cy.wait("@prefetchValues").then(request => {
-              const firstScoreRow = request.response.body;
-
-              actionForm().within(() => {
-                assertScoreFormPrefilled(firstScoreRow);
+    describe(
+      "in modal",
+      // These tests time out frequently in CI on `POST /api/dataset`
+      { viewportHeight: 1200, requestTimeout: 10000 },
+      () => {
+        const permissionLevels = [
+          {
+            name: "admin",
+            permissionFn: asAdmin,
+          },
+          {
+            name: "normal",
+            permissionFn: asNormalUser,
+          },
+        ];
+
+        permissionLevels.forEach(({ name, permissionFn }) => {
+          it(`should be able to run update and delete actions when enabled for a ${name} user`, () => {
```

Comment:
We should use `it.forEach` instead (for better console output).
I'll submit a PR with it shortly.